neqo-common/src/datagram.rs

```diff
@@ -83,6 +77,12 @@ impl std::fmt::Debug for Datagram {
     }
 }
 
+impl From<Datagram> for Vec<u8> {
+    fn from(datagram: Datagram) -> Self {
+        datagram.d
+    }
+}
+
```

Comment:
Is this part of the move or another change?

Reply:
The `udp` module needs to convert a `Datagram` to a `Vec`. Previously this would have been done via the `pub(crate)` `Datagram::into_data`.

Given that `udp` module is moved outside of `neqo-common` crate now, `pub(crate)` `into_data` is changed into a `pub` `From<Datagram> for Vec<u8>` implementation. See also https://github.com/mozilla/neqo/pull/1695#discussion_r1508330175.

Long story short, this just exposes `into_data` beyond the crate and implements it as a `From` instead.